File: three.js
function maxProduct(nums) {
    let max = 0;
    let sec = 0;
    let min = 0;
    let secmin = 0;
    nums.forEach(function (x) {
        if (x > max) {
            max = x;
        };
    });
    nums.forEach(function (x) {
        if (max > x && x >= sec) {
            sec = x;
        };
    });
    nums.forEach(function (x) {
        if (x < min) {
            min = x;
        };
    });
    nums.forEach(function (x) {
        if (min < x && x < secmin) {
            secmin = x;
        };
    });
    if (max * sec > min * secmin) {
        console.log(max * sec);
    } else {
        console.log(min * secmin);
    };
}

maxProduct([5, 20, 2, 6]);
maxProduct([10, -20, 0, 3]);

File: three.py
def maxProduct(nums):
    max = 0
    sec = 0
    min = 0
    secmin = 0
    for each in nums:
        if each > max:
            max = each
    for each in nums:
        if max > each >= sec:
            sec = each
    for each in nums:
        if each < min:
            min = each
    for each in nums:
        if min < each < secmin:
            secmin = each
    if max*sec > min*secmin:
        print(max*sec)
    else:
        print(min*secmin)


maxProduct([5, 20, 2, 6])  # 得到 120 因為 20 和 6 相乘得到最大值
maxProduct([10, -20, 0, 3])  # 得到 30 因為 10 和 3 相乘得到最大值
